### DISCUSSION TITLE
Codepspace cold-start startup time

### DISCUSSION THREAD

**Discussion by serdar** [community/community#3856] [Codespaces] [answered]
Hello,

When opening a Codespace that has been dormant for a while (a cold-start perhaps) takes minutes to reach a useable state. The repo is quite small comprising of only a handful of files.

This is against a private repository if that has any impact on the startup time. Please let me know what other information I can provide to assist.

Thanks

**Comment by sr229** [comment]
Sounds like this might be your internet rather than Codespaces. At average on a large repository, like Kubernetes, it only takes ~10-20s to allocate a Codespace for you.

**Reply by serdar** [reply]
Not ruling it out, although 1Gbps fiber should be enough for a tiny repo.

**Comment by 2percentsilk** [comment, marked as answer]
Hi @serdar,

We've made a ton of perf improvements over the last few weeks - are you still running into this? 

Thanks!

**Reply by serdar** [reply]
Thank you for updating me. I just got around to trying this again and the difference is night and day! Thanks again!
